django/thunderstore/repository/package_upload.py

```diff
@@ -151,8 +149,12 @@ def save(self, *args, **kwargs):
         self.instance.description = self.manifest["description"]
         self.instance.readme = self.readme
         self.instance.file_size = self.file_size
+
+        # Permissions and existence already checked when validating manifest
+        identity = UploaderIdentity.objects.get(name=self.manifest["author_name"])
```

Comment:
Should this try to get the same object the manifest sanitiser uses or is it fine it just fetch it?

Reply:
I'd make sure it's the same instance, so either pass it in as a key word argument to the serializer or get it from there.

I would probably pass it in to the serializer, and in the serializer, validate that the manifest author name matches the uploader identity name that got passed in as a keyword argument, and that the user has permissions there.